osquery/tables/system/windows/windows_update_history.cpp

```diff
@@ -0,0 +1,223 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#include <string>
+
+#include <atlbase.h>
+#include <windows.h>
+#include <wuapi.h>
+
+#include <osquery/core/core.h>
+#include <osquery/core/tables.h>
+#include <osquery/logger/logger.h>
+#include <osquery/utils/conversions/windows/strings.h>
+#include <osquery/utils/conversions/windows/windows_time.h>
+
+namespace osquery {
+namespace tables {
+
+HRESULT populateRow(IUpdateHistoryEntry* entry, Row& r);
+
+QueryData genUpdateHistory(QueryContext& context) {
+  QueryData results;
+
+  CComPtr<IUpdateSearcher> updateSearcher;
+  HRESULT hr = updateSearcher.CoCreateInstance(CLSID_UpdateSearcher);
+  if (FAILED(hr)) {
+    TLOG << "Failed to instantiate IUpdateSearcher";
```

Comment:
The logging should be used sparingly; it can probably be removed once this function makes use of `Expected<Value, Error>` and replaced with error codes